source/ceylon/net/http/server/MatcherRule.ceylon

```diff
@@ -16,6 +17,22 @@ shared abstract class Matcher() {
             => Or(this, other);
 }
 
+"Matcher to leverage Undertow's template mechanism for path templates and path parameters.
+ It should be given to an Endpoint without combining it with other matchers.
+ 
```

Comment:
Going to Endpoint(Matcher|TemplateMatcher path) would prevent it being a subclass of HttpEndpoint (and EndpointBase). This hierarchy (and class Endpoints as well as how it's used) stands a bit in the way of cleanly implementing TemplateCeylonRequestHandler and its integration in the DefaultServer.

Wouldn't that be a change too large for now?